pkg/common/common.go

```diff
@@ -41,7 +41,7 @@ const (
 type KuadrantPolicy interface {
 	client.Object
 	GetTargetRef() gatewayapiv1alpha2.PolicyTargetReference
```

Comment:
@adam-cattermole are you considering using the gatewayapiv1 ParentRef? e.g. https://github.com/Kuadrant/kuadrant-operator/compare/main...laurafitzgerald:kuadrant-operator:gateway-api-v1#diff-d9c7dc5384c33c14a24357bf23661864f2779ee1af0b7212050a6a79d0df65afR42

Reply:
We updated to using ParentRef in MGC so this would block us from consuming this version.

Reply:
Looking at the spec for [ParentRef](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.ParentReference) it looks to only support `Gateway` and `Service` but we need to support policies targetting an `HTTPRoute`. The description for [`PolicyTargetReference`](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1alpha2.PolicyTargetReference) also seems to better describe what we're using it for.

Reply:
It is possible to continue to use targetRef but the CRDs for v1.0.0 seem to have parentRef replace targetRef. So in MGC I've gone ahead and updated to using parentRef in the case of TLSPolicy and DNSPolicy. The error I'm facing is ```cannot use obj.(*v1alpha1.DNSPolicy) (comma, ok expression of type *"github.com/Kuadrant/multicluster-gateway-controller/pkg/apis/v1alpha1".DNSPolicy) as "github.com/laurafitzgerald/kuadrant-operator/pkg/common".KuadrantPolicy value in argument to policy.GetParentRefValueFromPolicy: *"github.com/Kuadrant/multicluster-gateway-controller/pkg/apis/v1alpha1".DNSPolicy does not implement "github.com/laurafitzgerald/kuadrant-operator/pkg/common".KuadrantPolicy (missing method GetTargetRef)
make: *** [vet] Error 1
``` 
So we may need to decouple our use of kuadrant-operator in this case

Reply:
The policy types don't seem to have made it into V1 however as it is a spec we are following I think it is fine to use the types we are currently using https://github.com/kubernetes-sigs/gateway-api/blob/0c3928466dc411788d03b1d3a21afefc482f7774/apis/v1alpha2/policy_types.go#L67